store/sqlstore/plugin_store.go

```diff
@@ -77,6 +77,11 @@ func (ps SqlPluginStore) CompareAndSet(kv *model.PluginKeyValue, oldValue []byte
 		return ps.CompareAndDelete(kv, oldValue)
 	}
 
+	// Expired entries should be treated as if they don't exist
```

Comment:
Should we instead only try this if `oldValue == nil` and we intend to update? That would remove one (redundant) in lots of cases.